testing/multi-env-test-engine/src/main/java/org/projectnessie/junit/engine/MultiEnvTestEngine.java

```diff
@@ -38,23 +34,39 @@
  * This is a JUnit5 Test Engine that delegates test discovery to {@link JupiterTestEngine} and
  * replicates the discovered tests for execution in multiple test environments.
  *
+ * <p>When multiple {@link MultiEnvTestExtension}s are applied to the same test, performs a cartesian
+ * product of available environments and their IDs.
+ *
+ * <p>For example:
+ * <ul>
+ *   <li>MultiEnvTestExtension segmentA with IDs 1, 2, 3</li>
+ *   <li>MultiEnvTestExtension segmentB with IDs 1, 2</li>
+ *   <li>MultiEnvTestExtension segmentC with ID 1</li>
+ * </ul>
+ * will result in the following IDs:
+ * <ul>
+ *   <li>[engine:nessie-multi-env][segmentA:1][segmentB:1][segmentC:1]</li>
+ *   <li>[engine:nessie-multi-env][segmentA:1][segmentB:2][segmentC:1]</li>
+ *   <li>[engine:nessie-multi-env][segmentA:2][segmentB:1][segmentC:1]</li>
+ *   <li>[engine:nessie-multi-env][segmentA:2][segmentB:2][segmentC:1]</li>
+ *   <li>[engine:nessie-multi-env][segmentA:3][segmentB:1][segmentC:1]</li>
+ *   <li>[engine:nessie-multi-env][segmentA:3][segmentB:2][segmentC:1]</li>
+ * </ul>
+ *
  * <p>Actual test environments are expected to be managed by JUnit 5 extensions, implementing the
  * {@link MultiEnvTestExtension} interface.
  */
 public class MultiEnvTestEngine implements TestEngine {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MultiEnvTestEngine.class);
-  private static final boolean FAIL_ON_MISSING_ENVIRONMENTS =
-      !Boolean.getBoolean("org.projectnessie.junit.engine.ignore-empty-environments");
+  private static final MultiEnvExtensionRegistry FILTER_REGISTRY = new MultiEnvExtensionRegistry();
```

Comment:
I'd prefer to keep the old lower case name... or perhaps `globalRegistry` if you want to make it easier to distinguish from `currentRegistry`.

Reply:
It was not needed. But I found a couple interesting things:
- The registry needs to be cleared between each test. I found tests like `bothOnJunitJupiter` would fail if run alone, but would pass when the whole `TestMultiEnvTestEngine` class was run. Clearing the registries between tests exposes this issue.
- The filter needs its own registry in case multi-env tests are selected on other engines (also see `bothOnJunitJupiter`)